### ISSUE TITLE
infoText refresh

### ISSUE THREAD

**Issue by neorana** [istvan-ujjmeszaros/bootstrap-duallistbox#117]
Hi, Good day,
I need to load the source data of dual listbox based on selected dropdown.
I was able to append the source of list box dynamically thru ajax, however, I’m having difficulty of refreshing the infoText for both boxes.

Note:
moveOnSelect: true
infoText should display info based on the number of records that I append.
Please help...Thanks a lot!
Any help will highly be appreciated.

Neo

**Comment by subprime** [comment]
Hi @neorana,
please provide more informations / code snippets about your problem. 
Currently i can not follow where problem is.

subprime

**Comment by neorana** [comment]
Hi @subprime 
Thanks for your reply, I was able to find a solution on this link http://stackoverflow.com/questions/19827177/bootstrap-dual-listbox-problems-while-populate-via-javascript.

Before this was my approach on success via ajax;

 $("#bootstrap-duallistbox-nonselected-list_GameSortListSelected").html(""); // clear non-selected before appending new list
  $("#bootstrap-duallistbox-selected-list_GameSortListSelected").html(""); //clear selected

then append it directly thru select attributes

$('#bootstrap-duallistbox-nonselected-list_GameSortListSelected').append($('<option>').text(value.GameName).val(value.GameId));

Now;
 $('#duallist').empty();
 dualList.trigger('bootstrapDualListbox.refresh');
getGames(); -- function to append the records
then call dual list refresh again
 dualList.trigger('bootstrapDualListbox.refresh', true);

But somehow, the sorting was inconsistent upon httppost.
Note: sortByInputOrder: true
The data-index order was not consistent on httppost. I need the original input order of selected records by the user to save it as is in the database.

Does it somehting to do with this function?
 function sortOptionsByInputOrder(select){
    var selectopt = select.children('option');

    selectopt.sort(function(a,b){
      var an = parseInt(a.getAttribute('data-sortindex')),
          bn = parseInt(b.getAttribute('data-sortindex'));

          if(an > bn) {
             return 1;
          }
          if(an < bn) {
            return -1;
          }
          return 0;
    });

    selectopt.detach().appendTo(select);
  }

Please help...thanks a lot
